dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/EncodingUtils.java

```diff
@@ -479,31 +394,33 @@ public static String getJvmConverter(String encoding) {
             return encoding;
         }
     }
-    
-	 /**
+
+    public static boolean isCharsetSupported (String charset){
+        return getCharsetForName(charset) != null;
+    }
+
+    /**
      * Tests whether the specified charset is supported on the server
      *
      * @param String The charset we want to test
      *
      * @return boolean indicating if supported
      */
     // rewritten as part of PK13492
-    public static boolean isCharsetSupported (String charset){
-        Boolean supported = (Boolean) supportedEncodingsCache.get(charset);
-        if(supported != null){
-            return supported.booleanValue();
-        }
-        try{
-            new String (TEST_CHAR, charset);
-            supportedEncodingsCache.put(charset, Boolean.TRUE);
-        }catch (UnsupportedEncodingException e){
-            if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable (Level.FINE)) {
-                logger.logp(Level.FINE, CLASS_NAME,"isCharsetSupported", "Encountered UnsupportedEncoding charset [" + charset +"]");
+    public static Charset getCharsetForName(String name) {
+        Charset charset = supportedEncodingsCache.get(name);
+        if (charset == null) {
+            try {
+                charset = Charset.forName(name);
+                supportedEncodingsCache.put(name, charset);
+            } catch (UnsupportedCharsetException e) {
+                if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
```

Comment:
Should this log be outside before the return? Otherwise, it will display once on the first set.

Reply:
I updated it to move the trace to be outside of the catch block.